src/ecalc/libraries/libecalc/common/libecalc/common/utils/rates.py

```diff
@@ -230,6 +232,27 @@ def __setitem__(
                     f"Could not update timeseries, Combination of indices of type '{type(indices)}' and values of type '{type(values)}' is not supported"
                 )
 
+    def __reindex_time_vector__(
+        self,
+        new_time_vector: Iterable[datetime],
+        fillna: Union[float, str] = 0.0,
+    ) -> np.ndarray:
+        """Based on a consumer time function result (EnergyFunctionResult), the corresponding time vector and
+        the consumer time vector, we calculate the actual consumer (consumption) rate.
+        """
+        new_values: DefaultDict[datetime, Union[float, str]] = defaultdict(float)
```

Comment:
Why can there a a string-value here?

Reply:
This reindex function is a pure copy of the reindex_time_vector in legacy_consumer/component.py (also used to reindex compressors) - have not checked in details, so not sure why string is allowed here.

Reply:
ok. Difficult to know if it was a reason in the past, or if it is still relevant. I think it might have been at the time when units were provided for each value, but not sure. I do not see any problems with that, but would be nice to know why. We should also ahve test for that wrt. regression, since if we change the functionality, it may easily break the "string-part" of reindexing imo, because one forgets to think about that part.